ansible/roles/test/tasks/bgp_neighbor.yml

```diff
@@ -0,0 +1,53 @@
+# Test file to verify that the bgp cli commands accurately work
+# Does the following
+#  - Determines all the BGP ipv4 and ipv6 neighbors
+#  - For each neighbor the test does the following
+#     - Shuts down the neighbor
+#     - Confirms the neighbor is down
+#     - Brings up the neighbor
+#     - Confirms the neighbor comes back up
+#
+# Test fails if any neighbors don't go down or come up as expected
+
+# Verify ipv4 operation
+- name: Execute Show Command
+  shell: show bgp ipv4 summary json
+  register: output
+
+- name: convert output to json and extract ipv4 bgp neighbors
+  set_fact:
+    bgp_neighbors: "{{ (output.stdout|from_json).ipv4Unicast.peers }}"
+
+- name: create a list of all the ipv4 bgp neighbors 
+  set_fact:
+    bgp_neighbor_list: "{{ bgp_neighbors.keys() | list }}"
+
+- name: verify ipv4 bgp neighbors are 'Established' or 'Active'
```

Comment:
Don't we expect all of them to be in established state? I feel, even if we have to wait for few seconds, we should wait and expect them to be in established state. Assuming that all neighbors were in established state before this exercise!

Reply:
I modified the code to stop the test if they aren't all established state to begin with.